cilium-cli/connectivity/check/logging.go

```diff
@@ -78,6 +81,60 @@ func (ct *ConnectivityTest) Log(a ...interface{}) {
 	fmt.Fprintln(ct.params.Writer, a...)
 }
 
+// ownedScenario represents a piece of logic in the testsuite with a
+// corresponding filepath that indicates ownership (via CODEOWNERS).
+// It is used to inform developers who they may consult in the event that a
+// test fails without a clear indication why.
+type ownedScenario interface {
+	Name() string
+	FilePath() string
+}
+
+var defaultTestOwners = defaultScenario{
+	ScenarioBase: NewScenarioBase(),
+}
+
+type defaultScenario struct {
+	ScenarioBase
+}
+
+func (s defaultScenario) Name() string {
+	return "cli-test-framework"
+}
+
+func (ct *ConnectivityTest) LogOwners(o ownedScenario) {
+	if !ct.params.LogCodeOwners {
+		return
+	}
+
+	rule, err := ct.CodeOwners.Match(o.FilePath())
+	if err != nil || rule == nil || rule.Owners == nil {
+		ct.Fatalf("Failed to find CODEOWNERS for test scenario. Developer BUG?",
+			"name=%s path=%s err=%s", o.Name(), o.FilePath(), err)
+		return
+	}
+
+	var workflowOwners []codeowners.Owner
+	// Example: octocat/hello-world/.github/workflows/my-workflow.yml@refs/heads/my_branch
+	ghWorkflow := os.Getenv("GITHUB_WORKFLOW_REF")
```

Comment:
For workflows that run `cilium connectivity test` from a github action, we would need to somehow retain a sensible workflow reference across that boundary. Not a blocker in my eyes but may require a little extra attention.

Reply:
This should be working correctly now.